Add new sideloadDataFn() to withContent & pageNode calls

B77Mills opened the pull request:
Add new sideloadDataFn to be passed into pageNode middleware.  This allows for additional data to be loaded along side the standard calls.

B77Mills committed:
Add new sideloadDataFn call to withContent & pageNode calls

Add new sideloadDataFn to be passed into pageNode middleware.  This allows for additional data to be loaded along side the standard graphql response.

solocommand commented on packages/marko-web/middleware/page-node/index.js:
```diff
@@ -9,21 +9,34 @@ class PageNode {
+      } else {
+        this.promise = this.apolloClient.query({ query, variables })
+          .then(r => createNode(getAsObject(r, path)));
+      }
```
Use `isFn` helper from base-cms-utils. Create a dummy function and always execute it, no need to duplicate the logic and nest if statements.

Don't bind your `this` context to the function -- you should not be exposing any internals of the page node middleware to the consuming function.

Don't mix async/await and then chains -- just use async/await.

```js
const fn = isFn(this.sideloadedDataFn) ? this.sideloadedDataFn : () => {};
const [r, data] = Promise.all([
  this.apolloClient.query(...),
  fn(),
];
this.promise = createNode(getAsObject(r, path));
this.sideloadedData = data;
```

B77Mills committed:
Update how sideloadDataFn is implemented

Check to see if this.sideloadDataFN is a function if it isnt set it to a simple function that returns an empty object.  Update calls after that to always set sideloadedData.

B77Mills committed:
clean and correctly setting the sideloadDataFn with binding
